### DISCUSSION TITLE
Should symbol picker opens at current document position?

### DISCUSSION THREAD

**Discussion by mrnossiom** [helix-editor/helix#10083] [Ideas] [answered]
Hi 👋🏻 

To quickly navigate symbols in a document, it would be great if the symbol picker would set your default position in the table to the closest symbol to your cursor. This makes navigation to upper and lower items easier, though doesn't modify the search.

**EDIT**: Also, more than navigation (which can be achieved with bracket shortcuts), it makes visibility of surrounding code easier (You can easily check order of close items in the files).

To me, this would go in the same way as #8875. Also, this could be implemented in the #9647 Picker V2 PR (which is done by @the-mikedavis).

I'd be happy to discuss here or elaborate my idea.

Also, I'm willing to do a PR. It seems that you would need to pass a new `initial_position` argument when creating the `Picker` to set it's `cursor` property, which sets the initial `TableState`'s `selected`. The issue to fix with this approach is that when searching, the selected row is already ahead which can be annoying.

**Comment by the-mikedavis** [comment]
This is an interesting idea. I'm trying it locally on my driver branch in https://github.com/the-mikedavis/helix/commit/3aec6d157051709a6130151177503ece5c6a7aa3. I'd like to avoid putting more changes into #9647 since it's already sprawling. The changes to do this can be distinct from the changes in that PR anyways

**Reply by the-mikedavis** [reply]
Making `cursor` `pub` is a hack though. To do that properly we would want to add a new function to Picker that takes a `&mut self` and sets the cursor and ensures that the value is in bounds

**Reply by mrnossiom** [reply]
I've been talking with @pascalkuthe on Matrix, I've adapted a bit of my discussion around cursor behaviour when you start typing for fuzzy matching.

When first typing your fuzzy search in the picker, your cursor looses meaning because everything is filtered and reordered.
Resetting the cursor would make sense to me. I still don't know if resetting the cursor would make sense on every keystroke, or just when you start filling the empty search. When there is an ongoing search, adding a character may filter only a few entries and not make lose meaning to the cursor.

I'll experiment with your driver branch

**Reply by pascalkuthe** [reply]
Resting everytime the prompt changes probably makes sense

**Reply by mrnossiom** [reply]
I've just tried, and first impressions are:
— It feels weird to sometimes have your cursor start at the end of the list, maybe that would not feel that weird if the picker scrolled instead of paginating.
— It doesn't have the effect I expected, it's not that nice to navigate functions, nor it gives a good overview of the current file organization.

**Reply by mrnossiom** [reply, marked as answer]
With the new picker out (which is fantastic btw!) and no real traction for this, I'm closing.

**Reply by ivanrg99** [reply]
I would love to see this implemented. Does the new picker do this?

**Reply by YPares** [reply]
It doesn't. I guess it won't be implemented, which is a bit sad because that's really the last missing piece IMHO.
